fit-vids.js

```diff
@@ -1,27 +1,36 @@
-class FitVids extends HTMLElement {
-  connectedCallback() {
-    const videoSources = ['iframe[src*="youtube"]', 'iframe[src*="vimeo"]'];
-    this.style.display = "block";
+export class FitVids extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: "open" });
+    this.shadowRoot.innerHTML = `
+      <style>
+        :host {
+          display: block;
+        }
+
+        ::slotted(iframe) {
+          aspect-ratio: var(--w, 16) / var(--h, 9);
+          height: auto;
+          max-width: 100%;
+          width: 100%;
+        }
+      </style>
+      <slot></slot>
```

Comment:
When moving to a `slot` you'll also have access to the [`slotchange`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/slotchange_event) event which means you can be resilient to the `iframe` changing over time and being applied through multiple slots by leveraging `slot.assignedElements({flatten: true})` rather than `this.querySelectorAll()`.

Reply:
Leveraging `slotchange` had crossed my mind. But if I'm modifying `slot` contents inside `slotchange` does that risk an infinite loop? 

It's probably just a matter of updating `connectedCallback()` → `_handleSlotChange()` and adding the eventListener tho. Easy update if it doesn't loop.